### ISSUE TITLE
[BUG] MSVC dispatch table injection fails to compile

### ISSUE THREAD

**Issue by erikzenker** [erikzenker/hsm#155]
**Describe the bug**
MSVC fails to compile `if constexpr` branches even if the condition for the branch does not evaluate to true.

This is needed to sm.h:
```c++
  if constexpr (has_anonymous_transition(rootState)) {
  ...
  auto& result = get_dispatch_table_entry(event, region);
  ...
  }
```

**To Reproduce**
Compiles on clang: https://godbolt.org/z/e3KcYv
Does not compile on msvc: https://godbolt.org/z/ab3bdPj91

**Expected behavior**
Compiles.

**Additional context**
A workaround could be to add some kind of a dummy next state for the none event.

**Comment by erikzenker** [comment]
[windows-ci-error.log](https://github.com/erikzenker/hsm/files/7611702/windows-ci-error.log)

**Comment by erikzenker** [comment]
Fixed by #173
